.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+name: CI and Lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run Prettier
+        run: npm run lint:prettier
+
+      - name: Lint changed files
+        if: github.event_name == 'pull_request'
```

Comment:
Since this step does the actual linting, and it only runs on 'pull_request', we should probably do one of the following:
- Either remove the `push` line from the `on` section at the top of the file
- Or, allow this linting to run on both push and pull request

What do you think?

Reply:
Thanks for pointing this out! After reviewing, I agree that having the push trigger doesn't make much sense here since this step is focused on linting for pull requests. I'll update the workflow to remove the push line from the on section, ensuring that this workflow only runs on pull requests.